### PR TITLE
Channel iteration enhancements

### PR DESCRIPTION
**Pull request by Zoomah**
### Additions and changes to iterate through and work with channels

ADD
- [x] `{categories}` guild->channels (filtered to categories): Array\<Channel>
- [x] `{channels}` guild->channels: Array\<Channel>
- [x] `{channels;ID}` category ID->channels: Array\<Channel>
- [x] `{category;ID}` channel->category: String (empty string if no category)
---
CHANGE
- [x] `{channelname;[ID]}` channel->name : String
- [x] add Alias: `categoryname`
- [x] `{channelid;[NAME]}` channel->id : String 
- [x] add Alias: `categoryid`
- [x] `{channelpos;[ID]}` channel->position : String
- [x] add Alias: `categorypos`
---
ADD
- [x] `{istext;[ID]}` channel-> type == text : Bool
- [x] `{isvoice;[ID]}` channel-> type == voice : Bool
- [x] `{iscategory;ID}` channel-> type == category : Bools
---
CHANGE
- [x] `{isnsfw}` add quiet option
---
Question for testing: Is `{indexof;{channels};ID}` the same as `{channelpos;ID}` ?

### CODE REVIEW

**Comment by byparallax** [comment]
You make my Pee Pee feel funny.

File: src/tags/guildchannels.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * @Last Modified by: stupid cat
+ * @Last Modified time: 2018-04-27 13:14:24
+ *
+ * This project uses the AGPLv3 license. Please read the license file before using/adapting any of the code.
+ */
+
+const Builder = require('../structures/TagBuilder');
+
+module.exports =
+    Builder.APITag('guildmembers')
```
Comment (Titansmasher) [inline src/tags/guildchannels.js]: Wrong name

File: src/tags/guildchannels.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        .withDesc('Returns an array of user IDs of the channels on the current guild.')
```
Comment (Titansmasher) [inline src/tags/guildchannels.js]: user ids of the channels? 😜

File: src/tags/channelpos.js [hunk 3]
```diff
@@ -11,11 +11,20 @@ const Builder = require('../structures/TagBuilder');
...
+        .withArgs(a => [a.optional('channelid')])
+        .withDesc('Returns the position of the current channel. If no channelid is given, the current channels position will be returned.')
         .withExample(
             'This channel is in position {channelpos}',
             'This channel is in position 1'
         )
         .whenArgs(0, async (_, context) => context.channel.position)
+        .whenArgs(1, async (_, context, args) => {
+            let i = 0;
+            context.guild.channels.foreach(g => {
```
Comment (Titansmasher) [inline src/tags/channelpos.js]: Would it be possible to use indexof here rather than a loop?

**Comment by Saederup92** [comment]
@byparallax Eat da poo poo

File: src/tags/channelpos.js (hunk 3 above)
Comment (loliprotector) [reply src/tags/channelpos.js]: Feel like it should be renamed to `{channels}` for consistency with its closest relative: `{roles}`

- [ ] Add `<channel> [quiet]` argument in `{channelid}`
    - Similar to `{roleid;<role>;[quiet]}`

- [ ] Rename `{channelpos}` to `{channels}`
    - Similar to `{roles}`


```cs
{indexof;
    {roles};
    {roleid;roleName}
}
```
> To find the specific position of a role.

```cs
{indexof;
    {channels};
    {channelid;channelName}
}
```
> To find the specific position of a channel.

**Comment by loliprotector** [comment]
Feel like `{channelpos}` should be renamed to `{channels}` for consistency with its closest relative: `{roles}`

- [ ] Add `<channel> [quiet]` argument in `{channelid}`
    - Similar to `{roleid;<role>;[quiet]}`

- [ ] Rename `{channelpos}` to `{channels}`
    - Similar to `{roles}`


```cs
{indexof;
    {roles};
    {roleid;roleName}
}
```
> To find the specific position of a role.

```cs
{indexof;
    {channels};
    {channelid;channelName}
}
```
> To find the specific position of a channel.

File: src/tags/guildchannels.js (hunk 1 above)
Comment (Zoomah) [reply src/tags/guildchannels.js]: oops... yus!

File: src/tags/guildchannels.js (hunk 2 above)
Comment (Zoomah) [reply src/tags/guildchannels.js]: err... yeah

**Comment by Zoomah** [comment]
Unfortunately, thats not what Eris gives us... The code would be a LOT more complicated to do it on that scheme

**Comment by Zoomah** [comment]
OOOkie... changed a few things... what you suggested should be possible now @loliprotector

**Pull request closed without merging**